agasc/agasc.py

```diff
@@ -132,6 +137,10 @@ def get_agasc_cone(ra, dec, radius=1.5, date=None, agasc_file=None,
     # Possibly expand initial radius to allow for slop due proper motion
     rad_pm = radius + (0.1 if pm_filter else 0.0)
 
+    global RA_DECS
```

Comment:
Since this is a core package, let's make this a little more idiomatic and encapsulated.  

Revert the original `RA_DECS = RaDec()`.  Then add a setter to that class (`@agasc_file.setter`) which allows setting the `agasc_file` and also clears the cache (by `del self.ra; del self.dec`) (but only if the new agasc_file is different from the current one).  Then you need `@ra.deleter` and `@dec.deleter`, which delete the `_ra` or `_dec` attr if it exists.  And finally:
```
if agasc_file is not None:
    orig_agasc_file = RA_DECS.agasc_file
    RA_DECS.agasc_file = agasc_file

...

if agasc_file is not None:
    RA_DECS.agasc_file = orig_agasc_file
```
This will be slow because it initializer `RA_DECS` for each call with a custom `agasc_file`.  But if you want to work in general with a different file you could explicitly do `RA_DECS.agasc_file = my_agasc_file` at the beginning.